libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController.m

```diff
@@ -72,25 +73,45 @@ - (void)viewDidLoad {
     self.view.autoresizesSubviews = YES;
     self.view.autoresizingMask = UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
     self.view.clipsToBounds = YES;
-    if(self.showNavbar){
+    if (self.showNavbar){
         [self setupNavigationBar];
     } else {
         self.navigationController.navigationBarHidden = YES;
     }
-    [self layoutWebView];
     
+    [self.view addSubview:_oauthView];
+
     SFBiometricAuthenticationManagerInternal *bioAuthManager = [SFBiometricAuthenticationManagerInternal shared];
+    BOOL showBioAuthButton = [bioAuthManager showNativeLoginButton];
+
+    if (showBioAuthButton) {
+        UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+        [button setTitle:[SFSDKResourceUtils localizedString:@"biometricLoginButton"] forState:UIControlStateNormal];
+        [button addTarget:self action:@selector(presentBioAuthAction:) forControlEvents:UIControlEventTouchUpInside];
+        self.biometricButton = button;
+        [self.view addSubview:self.biometricButton];
+    }
+    
     if (bioAuthManager.locked && bioAuthManager.hasBiometricOptedIn) {
         [bioAuthManager presentBiometricWithScene:self.view.window.windowScene];
     }
 }
 
-- (CGFloat) belowFrame:(CGRect) frame {
-    return frame.origin.y + frame.size.height;
-}
-
-- (void)viewDidLayoutSubviews {
-    [super viewDidLayoutSubviews];
+- (void)viewWillLayoutSubviews {
+    CGFloat heightOffsetMultiplier = _biometricButton ? 0.9 : 1.0;
+    CGFloat bottomOffset = _biometricButton ? self.view.safeAreaInsets.bottom : 0;
+    
+    // Web view
+    CGFloat x = 0;
+    CGFloat y = self.view.safeAreaInsets.top;
```

Comment:
This is the part where I'm not sure how it would play with [this line](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/030dec5e2ab53574ff011365952d93a10f4e5e86/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Login/SFLoginViewController.m#L195) where the nav bar is added to the view directly, so we could try tracking that down or use the old nav bar calculation with the new `viewWillLayoutSubviews` and bottom inset pieces?

Reply:
My brain has sort of shut down for the weekend lol.  Are you saying we should hard code `y` to 0 instead?

Reply:
sounds good!